### DISCUSSION TITLE
Can't test against firefox after upgrade to v10. Getting "Still waiting to connect to Firefox, retrying in 1 second".

### DISCUSSION THREAD

**Discussion by JohnLukeBentley** [cypress-io/cypress#22117] [Questions and Help] [answered]
After an upgrade to Cypress 10.0.1 (which distinguishes, for the first time, End to End V Component testing) I've successfully migrated my project and can run End to End (e2e) tests in Chrome. But I can't run these in firefox ...

```powershell
> npm test

> testers@1.0.1 test
> npx cypress open


Still waiting to connect to Firefox, retrying in 1 second (attempt 18/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 19/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 20/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 21/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 22/62)
[etc ...]
```

To get that failed result I:

* (In effect) from the command line run ` npx cypress open`. That is, `npm test` references `scripts: { ..."test": "test": "npx cypress open"}} ...` in package.json.
* The Cypress test environment opens and I choose [E2E Testing]
* Firefox is selected. I choose [Start E2E Testing in Firefox]
* The Cypress test runner (browser) opens, with the browser body being blank and the "Still waiting to connect to Firefox, retrying in 1 second" messages being output to the command line (as above).

Windows 10 64bit Pro 21H1 (19043.1706)
Firefox 101.0 (64-bit).

The problem persists under cypress 10.0.2.

I have mentioned this at https://github.com/cypress-io/cypress/issues/18919#issuecomment-1144415625, but I post here for visibility.

Shall I file a bug report or is there some basic thing for me to check first? Can anyone else reproduce this problem?

**Comment by tomrobst** [comment]
Hi, 

I'm having the same issue as JohnLukeBentley on Cypress v10.0.3. 

When starting e2e tests on Firefox, the browser opens but does not start cypress and the console outputs: "Still waiting to connect to Firefox, retrying in 1 second..." etc

I have run Cypress with debugging enabled and it seems there is an issue finding the browser target:

2022-06-07T14:48:40.987Z cypress:server:browsers:browser-cri-client Attaching to target url about:blank
Still waiting to connect to Firefox, retrying in 1 second (attempt 25/62)
2022-06-07T14:48:40.988Z cypress:server:browsers:browser-cri-client error finding browser target, maybe retrying { delay: 1000, err: Error: Could not find url target in browser about:blank. Targets were [] at C:\Users\tomro\AppData\Local\Cypress\Cache\10.0.3\Cypress\resources\app\packages\server\lib\browsers\browser-cri-client.js:90:27 at processTicksAndRejections (node:internal/process/task_queues:96:5) at retry (C:\Users\tomro\AppData\Local\Cypress\Cache\10.0.3\Cypress\resources\app\packages\server\lib\browsers\browser-cri-client.js:39:20) at setupRemote (C:\Users\tomro\AppData\Local\Cypress\Cache\10.0.3\Cypress\resources\app\packages\server\lib\browsers\firefox-util.js:109:27)  }

Manually going to the baseURL in the opened browser window (in my case http://192.168.1.107:8081/__/#/specs) does open the cypress specs in Firefox and testing seems to work from this point, however the cypress command line log still fails to connect.

Please let me know if there are any other relevant logs which may be of use.

WIndows 11 Pro 10.0.22621
Firefox 101.0 (64-bit)
Cypress is running on WSL2

Thanks very much!
Tom
--

**Comment by JohnLukeBentley** [comment]
@tomrobst glad to learn I'm not the only one.

I too can confirm that when Cypress test runner (firefox browser) opens, and chokes, the address bar is empty (it displays "Search with Google or enter address"); and that manually putting in the right URL (in my case http://127.0.0.1:5500/__/#/specs) opens that cypress specs. And, like you, I can then run my *.cy.js spec without trouble.

So that choke point remains between hitting [Start E2E Testing in Firefox] and having the Cypress test runner (firefox browser) open up correctly.

Windows: Windows 10 Pro (64bit) 21H1, 19043.1706
Firefox: 101.0 (64-bit)
Cypress:  10.0.3

**Reply by tomrobst** [reply]
@JohnLukeBentley - it looks like firefox just isn't opening and going to the baseURL properly. 

There's now an open issue for this so hopefully it should be fixed soon! https://github.com/cypress-io/cypress/issues/22086

Tom
--

**Comment by JohnLukeBentley** [comment, marked as answer]
Thanks @tomrobst. My searching for that issue failed me :(.

So we can now close this thread in favour of issue https://github.com/cypress-io/cypress/issues/22086
